connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -139,25 +139,30 @@ public void execute() {
     }
 
     protected void iteration() {
-        long now = time.milliseconds();
-
-        // Maybe commit
-        if (!committing && now >= nextCommit) {
-            commitOffsets(now, false);
-            nextCommit += workerConfig.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-        }
+        long timeoutMs;
+        if (!context.offsetCommitDisabled()) {
+            long now = time.milliseconds();
+
+            // Maybe commit
+            if (!committing && now >= nextCommit) {
+                commitOffsets(now, false);
+                nextCommit += workerConfig.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
+            }
 
-        // Check for timed out commits
-        long commitTimeout = commitStarted + workerConfig.getLong(
+            // Check for timed out commits
+            long commitTimeout = commitStarted + workerConfig.getLong(
                 WorkerConfig.OFFSET_COMMIT_TIMEOUT_MS_CONFIG);
-        if (committing && now >= commitTimeout) {
-            log.warn("Commit of {} offsets timed out", this);
-            commitFailures++;
-            committing = false;
-        }
+            if (committing && now >= commitTimeout) {
+                log.warn("Commit of {} offsets timed out", this);
+                commitFailures++;
+                committing = false;
+            }
 
-        // And process messages
-        long timeoutMs = Math.max(nextCommit - now, 0);
+            // And process messages
+            timeoutMs = Math.max(nextCommit - now, 0);
+        } else {
+            timeoutMs = workerConfig.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
```

Comment:
Yeah, that's a little odd. If we don't do this, we may have to introduce another config that controls the poll timeout. 

Another thing I am thinking about is whether the global 'OFFSET_COMMIT_INTERVAL_MS_CONFIG' makes sense or not. We may want the connector to choose the commit interval. Maybe we can also implement other commit policy like size or number of messages which should also be controlled by the connector.